src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -1002,6 +1059,47 @@ private SnapshotMapColumn lookupMapColumn(Column toColumn, SnapshotMapTable mapT
         return null;
     }
 
+    private void softDeleteRows(List<String> softDeleteRowIds,
+                                String tableName,
+                                Dataset datarepoDataset,
+                                BigQuery bigQuery) {
+        BigQueryProject bigQueryProject = bigQueryProjectForDataset(datarepoDataset);
+        String projectId = bigQueryProject.getProjectId();
+        String softDeletesTableName = projectId + "." + datarepoDataset +  "." + createSoftDeletesTableName(tableName);
+
+        // TODO: Validate rowIDs exist in given table
+        String rowIdValues = softDeleteRowIds
+            .stream()
+            .map(rowId ->  String.format("'%s'", rowId))
+            .collect(Collectors.joining(","));
+
```

Comment:
i think this block can be replace with
        String rowIdValues = String.join(",", softDeleteRowIds);